googlemaps/matching-algo.js

```diff
@@ -0,0 +1,152 @@
+var googleMapsClient = require("@google/maps").createClient({
+  key: "AIzaSyBxp25VtQir86dy8rbjpQCAo4XtTMw8TzM",
+  Promise: Promise
+});
+
+
+//DEFINITIONS:
+// ABCD trip = trip from the driver's departure location to his arrival location 
+//             with two stopovers (pick up and drop off passenger)
+
+
+// Calculates the duration of the original trip posted by a user
+function calcUserDur(userTrip){
+  googleMapsClient.directions(
+    {
+      origin: [userTrip.startLocation.coordinates[1],userTrip.startLocation.coordinates[0]],
+      destination: [userTrip.endLocation.coordinates[1],userTrip.endLocation.coordinates[0]],
+      mode: "driving"
+    }
+  ).asPromise()
+  .then((response) => {
+    let userDur = response.json.routes[0].legs[0].duration.value
+    return userDur;
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+};
+
+
+function calcBestMatch(isDriver, userTrip, db){
+
+  let minDur = 99999999;
+  const bestMatch = {};
+
+  //for each trip in the array, we ask google map for the duration of the ABCD trip
+  var googleRequests =
+    db.map(currentTrip => {
```

Comment:
googleRequests doesn't change, right? If so, let's make it a const

Reply:
Bien vu l'artiste! (lol...sorry)